### PR TITLE
Fix the bugs detected by sonar

### PR DESCRIPTION
**Pull request by mmoqui**
Merge also the PR 481 in Silverpeas Component

### CODE REVIEW

File: core-library/src/main/java/org/silverpeas/core/index/indexing/parser/ps/PsParser.java [hunk 1]
```diff
@@ -90,6 +87,7 @@ public void outPutChar(Writer out, BufferedReader buffer) throws IOException {
           } else {
             out.write(charr);
           }
+          break;
         case '\n':
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/index/indexing/parser/ps/PsParser.java]: MAJOR Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1151)

File: core-library/src/main/java/org/silverpeas/core/admin/domain/driver/ldapdriver/LDAPTimeStampNDS.java [hunk 2]
```diff
@@ -41,6 +42,12 @@ public int compareTo(Object other) {
     return timeStamp.compareTo(((LDAPTimeStampNDS) other).timeStamp);
   }
 
+  @Override
+  public int hashCode() {
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/admin/domain/driver/ldapdriver/LDAPTimeStampNDS.java]: BLOCKER This class overrides "hashCode()" and should therefore also override "equals()". [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1206)

File: core-services/calendar/src/main/java/org/silverpeas/core/calendar/model/Priority.java [hunk 3]
```diff
@@ -25,18 +25,16 @@
 package org.silverpeas.core.calendar.model;
 
 import java.io.Serializable;
+import java.util.HashMap;
```
Comment (silveruser) [inline core-services/calendar/src/main/java/org/silverpeas/core/calendar/model/Priority.java]: MINOR Remove this unused import 'java.util.HashMap'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

File: core-services/calendar/src/main/java/org/silverpeas/core/calendar/model/Priority.java [hunk 4]
```diff
@@ -25,18 +25,16 @@
...
 
+import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.silverpeas.core.silvertrace.SilverTrace;
```
Comment (silveruser) [inline core-services/calendar/src/main/java/org/silverpeas/core/calendar/model/Priority.java]: MINOR Remove this unused import 'org.silverpeas.core.silvertrace.SilverTrace'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

File: core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/arraypanes/ArrayLine.java [hunk 5]
```diff
@@ -379,14 +375,40 @@ public int compareTo(final java.lang.Object other) {
...
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof ArrayLine)) {
+      return false;
+    }
+
+    final ArrayLine arrayLine = (ArrayLine) o;
```
Comment (silveruser) [inline core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/arraypanes/ArrayLine.java]: MAJOR Remove this unused "arrayLine" local variable. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1481)

File: core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java [hunk 6]
```diff
@@ -56,11 +56,6 @@
 @Named("openDocumentParser")
 public class OOParser implements Parser {
 
-  /**
-   * Constructor declaration
-   */
-  public OOParser() {
-  }
   private String tempFolder = null;
   private final String TMP_UNZIP_DIR = "tmpUnzipOpenOffice";
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java]: MAJOR Make this final field static too. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1170)
MAJOR Rename this field "TMP_UNZIP_DIR" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00116)

File: core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java [hunk 7]
```diff
@@ -56,11 +56,6 @@
...
   private final Namespace NS_OO = Namespace.getNamespace("office",
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java]: MAJOR Rename this field "NS_OO" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00116)

File: core-library/src/main/java/org/silverpeas/core/admin/domain/driver/ldapdriver/LDAPTimeStampMSAD.java [hunk 8]
```diff
@@ -49,6 +50,12 @@ else if (lTimeStamp < ((LDAPTimeStampMSAD) other).lTimeStamp)
       return 0;
   }
 
+  @Override
+  public int hashCode() {
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/admin/domain/driver/ldapdriver/LDAPTimeStampMSAD.java]: BLOCKER This class overrides "hashCode()" and should therefore also override "equals()". [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1206)

**Comment by silveruser** [comment]
SonarQube analysis reported 49 issues
* BLOCKER 2 blocker
* CRITICAL 1 critical
* MAJOR 33 major
* MINOR 13 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [PsParser.java#L134](https://github.com/Silverpeas/Silverpeas-Core/blob/bd1eef0399bd6bda7af4abf1ecb8fcadef24503a/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/ps/PsParser.java#L134): End this switch case with an unconditional break, return or throw statement. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS128)
1. MAJOR [HTMLParser2.java#L50](https://github.com/Silverpeas/Silverpeas-Core/blob/bd1eef0399bd6bda7af4abf1ecb8fcadef24503a/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/html/HTMLParser2.java#L50): Missing curly brace. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00121)
1. MAJOR [OOParser.java#L63](https://github.com/Silverpeas/Silverpeas-Core/blob/bd1eef0399bd6bda7af4abf1ecb8fcadef24503a/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java#L63): Rename this field "NS_DC" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00116)
1. MAJOR [OOParser.java#L65](https://github.com/Silverpeas/Silverpeas-Core/blob/bd1eef0399bd6bda7af4abf1ecb8fcadef24503a/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java#L65): Rename this field "NS_OOMETA" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00116)
1. MAJOR [OOParser.java#L67](https://github.com/Silverpeas/Silverpeas-Core/blob/bd1eef0399bd6bda7af4abf1ecb8fcadef24503a/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java#L67): Rename this field "NS_OOTEXT" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00116)
1. MAJOR [OOParser.java#L70](https://github.com/Silverpeas/Silverpeas-Core/blob/bd1eef0399bd6bda7af4abf1ecb8fcadef24503a/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java#L70): Make this final field static too. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1170)
1. MAJOR [OOParser.java#L70](https://github.com/Silverpeas/Silverpeas-Core/blob/bd1eef0399bd6bda7af4abf1ecb8fcadef24503a/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java#L70): Rename this field "TITLE" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00116)
1. MAJOR [OOParser.java#L71](https://github.com/Silverpeas/Silverpeas-Core/blob/bd1eef0399bd6bda7af4abf1ecb8fcadef24503a/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java#L71): Make this final field static too. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1170)
1. MAJOR [OOParser.java#L71](https://github.com/Silverpeas/Silverpeas-Core/blob/bd1eef0399bd6bda7af4abf1ecb8fcadef24503a/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java#L71): Rename this field "SUBJECT" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00116)
1. MAJOR [OOParser.java#L72](https://github.com/Silverpeas/Silverpeas-Core/blob/bd1eef0399bd6bda7af4abf1ecb8fcadef24503a/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java#L72): Make this final field static too. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1170)

File: core-library/src/main/java/org/silverpeas/core/admin/domain/driver/ldapdriver/LDAPTimeStampNDS.java (hunk 2 above)
Comment (mmoqui) [inline core-library/src/main/java/org/silverpeas/core/admin/domain/driver/ldapdriver/LDAPTimeStampNDS.java]: The equals() method is provided by the parent class

File: core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java [hunk 9]
```diff
@@ -56,30 +56,25 @@
...
-  private final String KEYWORD = "keyword";
+  private static final String TITLE = "title";
+  private static final String SUBJECT = "subject";
+  private static final String DESCRIPTION = "description";
+  private static final String INITIAL_CREATOR = "initial-creator";
+  private static final String KEYWORD = "keyword";
   // Open Office files needed for indexing
-  private final String contentFile = "content.xml";
-  private final String metaFile = "meta.xml";
+  private static final String contentFile = "content.xml";
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java]: MAJOR Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00115)

File: core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java [hunk 10]
```diff
@@ -56,30 +56,25 @@
...
+  private static final String metaFile = "meta.xml";
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java]: MAJOR Rename this constant name to match the regular expression '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00115)

**Comment by silveruser** [comment]
SonarQube analysis reported 29 issues
* BLOCKER 2 blocker
* CRITICAL 1 critical
* MAJOR 15 major
* MINOR 11 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. CRITICAL [PsParser.java#L134](https://github.com/Silverpeas/Silverpeas-Core/blob/6d9c4d0c9ae72963cce45a0b6e3d0d7dc411975b/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/ps/PsParser.java#L134): End this switch case with an unconditional break, return or throw statement. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS128)
1. MAJOR [OOParser.java#L96](https://github.com/Silverpeas/Silverpeas-Core/blob/6d9c4d0c9ae72963cce45a0b6e3d0d7dc411975b/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java#L96): The Cyclomatic Complexity of this method "parse" is 12 which is greater than 10 authorized. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. MAJOR [OOParser.java#L171](https://github.com/Silverpeas/Silverpeas-Core/blob/6d9c4d0c9ae72963cce45a0b6e3d0d7dc411975b/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java#L171): Move the "meta.xml" string literal on the left side of this string comparison. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1132)
1. MAJOR [OOParser.java#L171](https://github.com/Silverpeas/Silverpeas-Core/blob/6d9c4d0c9ae72963cce45a0b6e3d0d7dc411975b/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/odf/OOParser.java#L171): Move the "content.xml" string literal on the left side of this string comparison. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1132)
1. MAJOR [PsParser.java#L77](https://github.com/Silverpeas/Silverpeas-Core/blob/6d9c4d0c9ae72963cce45a0b6e3d0d7dc411975b/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/ps/PsParser.java#L77): The Cyclomatic Complexity of this method "outPutChar" is 26 which is greater than 10 authorized. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. MAJOR [PsParser.java#L84](https://github.com/Silverpeas/Silverpeas-Core/blob/6d9c4d0c9ae72963cce45a0b6e3d0d7dc411975b/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/ps/PsParser.java#L84): Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1151)
1. MAJOR [PsParser.java#L103](https://github.com/Silverpeas/Silverpeas-Core/blob/6d9c4d0c9ae72963cce45a0b6e3d0d7dc411975b/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/ps/PsParser.java#L103): Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1151)
1. MAJOR [PsParser.java#L111](https://github.com/Silverpeas/Silverpeas-Core/blob/6d9c4d0c9ae72963cce45a0b6e3d0d7dc411975b/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/ps/PsParser.java#L111): Reduce this switch case number of lines from 29 to at most 5, for example by extracting code into methods. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1151)
1. MAJOR [PsParser.java#L112](https://github.com/Silverpeas/Silverpeas-Core/blob/6d9c4d0c9ae72963cce45a0b6e3d0d7dc411975b/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/ps/PsParser.java#L112): Missing curly brace. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00121)
1. MAJOR [PsParser.java#L113](https://github.com/Silverpeas/Silverpeas-Core/blob/6d9c4d0c9ae72963cce45a0b6e3d0d7dc411975b/core-library/src/main/java/org/silverpeas/core/index/indexing/parser/ps/PsParser.java#L113): Extract the assignment out of this expression. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)

**Comment by silveruser** [comment]
SonarQube analysis reported 2 issues
* BLOCKER 2 blocker

Watch the comments in this conversation to review them.

File: core-war/src/main/java/org/silverpeas/web/jobstartpage/control/JobStartPagePeasSessionController.java [hunk 11]
```diff
@@ -980,7 +980,7 @@ public void deleteComponentInBin(String componentId) {
         m_BrothersComponents[j++] = theComponent;
       }
     }
-    Arrays.sort(m_BrothersComponents);
+    Arrays.sort(m_BrothersComponents, Comparator.comparing((ComponentInst::getOrderNum)));
```
Comment (silveruser) [inline core-war/src/main/java/org/silverpeas/web/jobstartpage/control/JobStartPagePeasSessionController.java]: MAJOR Remove those useless parentheses. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

**Comment by silveruser** [comment]
SonarQube analysis reported 1 issue
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request merged**